src/XMakeBuildEngine/Definition/Project.cs

```diff
@@ -2162,6 +2176,14 @@ internal bool SplitItemElementIfNecessary(ProjectItemElement itemElement)
             return true;
         }
 
+        internal bool ItemElementRequiresSplitting(ProjectItemElement itemElement)
+        {
+            var hasCharactersThatRequireSplitting = FileMatcher.HasWildcardsSemicolonItemOrPropertyReferences(itemElement.Include);
+            var hasMoreThanOneItem = Items.Count(i => i.Xml == itemElement) > 1;
```

Comment:
I wouldn't know what to write other than "this is how the code did it" ... I can scour the history and see who added the splitting code.